File: backend/app/schemas/auditoria.py
# app/schemas/auditoria.py
"""Esquema de salida del visor de auditoría (registro de cambios)."""

from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class AuditoriaOut(BaseModel):
    id: int
    tabla: str
    operacion: str                     # INSERT | UPDATE | DELETE
    registro_id: str | None = None
    actor_app: str | None = None       # "id|email" del usuario de la app
    actor_bd: str | None = None
    datos_antes: dict[str, Any] | None = None
    datos_despues: dict[str, Any] | None = None
    ocurrido_en: datetime

    model_config = ConfigDict(from_attributes=True)

File: movil/src/api/auditoria.ts
// src/api/auditoria.ts
// Visor del registro de cambios (auditoría, Ley 21.719). Solo RRHH/Admin.
// Espejo de frontend/src/features/auditoria/api.ts.

import { request } from '../services/http'

export interface RegistroAuditoria {
  id: number
  tabla: string
  operacion: 'INSERT' | 'UPDATE' | 'DELETE' | string
  registro_id?: string | null
  actor_app?: string | null      // "id|email" del usuario de la app
  actor_bd?: string | null
  datos_antes?: Record<string, unknown> | null
  datos_despues?: Record<string, unknown> | null
  ocurrido_en: string            // ISO datetime
}

export async function getAuditoria(
  tabla?: string,
  limite = 100,
): Promise<RegistroAuditoria[]> {
  const params = new URLSearchParams()
  if (tabla) params.set('tabla', tabla)
  params.set('limite', String(limite))
  return request<RegistroAuditoria[]>(`/auditoria?${params.toString()}`)
}
